WoWDatabaseEditor.Common/WDE.DbcStore/Providers/DBDProvider.cs

```diff
@@ -0,0 +1,47 @@
+﻿using DBCD.Providers;
+using System;
+using System.IO;
+using System.Net.Http;
+
+namespace WDE.DbcStore.Providers
+{
+    public class DBDProvider : IDBDProvider
+    {
+        private static Uri BaseURI = new Uri("https://raw.githubusercontent.com/wowdev/WoWDBDefs/master/definitions/");
+        private static string CachePath = "dbdCache/";
+        private HttpClient client = new HttpClient();
+
+        public DBDProvider()
+        {
+            if (!Directory.Exists(CachePath))
```

Comment:
I think it is better to write to user directory instead of current folder (application folder), since you may not have permission to write in the current directory (e.g. on MacOS) and also after every update, the cache will get lost.

To write in the user directory, add `IVirtualFileSystem` to the constructor and use its method using paths with a tilde in the beginning, i.e. `~/dbdCache/`. This path will be resolved to user local folder.

If some method is missing in `IVirtualFileSystem`, feel free to add it.